hw_diag/utilities/balena_supervisor.py

```diff
@@ -56,11 +56,24 @@ def get_device_status(self, key_to_return) -> str:
         LOGGER.info("Retrieving device status using Balena supervisor.")
 
         response = self._make_request('GET', '/v2/state/status')
-        if response is None or response.ok is False:
-            LOGGER.error("Device status request failed.")
-            raise Exception("Device status request failed")
+
+        error_msg = ""
+        if response is None:
+            error_msg = "Device status request failed. No response recieved."
+
+        elif response.ok is False:
```

Comment:
nit: `elif not response.ok` would be more idiomatic

Reply:
A matter of preference but per zen of python "explicit is better than implicit" and I've seen some code linting tools like python-black suggest comparisons like `is False` instead of `not value`.